I flubbed the commit on the last one, this one should be good (I hope)

brianb opened the pull request:
This patch replaces the previous one and includes the unit test changes

xiaoerge commented on src/main/java/com/xiaoerge/x12/message/MessageFormat.java:
```diff
@@ -0,0 +1,52 @@
+    
+    public MessageFormat() {
+	}
+    public MessageFormat(String s) throws MessageFormatException {
```
What do you think about removing this constructor? For couple reasons,  
A) I prefer the parser not throwing hard exception on incorrect message format. Since Segment.java and util classes are only logging warnings. 
B) This constructor method isn't tested for message parsing.

I'd say we remove it and expose setters for them.
The parser requires some knowledge of the incoming message anyway so knowing the proper terminators to set for message format should be fine.

brianb replied:
So the thing is, that the delimiters should be fully discoverable from the ISA segment itself without any advanced knowledge of the message format.  I'd be ok moving this to an auxiliary class, but I wouldn't want to lose the ability to detect the delimiters automagically.

xiaoerge commented on src/main/java/com/xiaoerge/x12/message/MessageFormat.java:
```diff
@@ -0,0 +1,52 @@
+    private String componentElementSeparator = "\\";
+    private String segmentTerminator = "~";
+    private String repetitionSeparator = "^";
+    private String suffix = null;
```
Test this case
http://stackoverflow.com/questions/4260723/concatenating-null-strings-in-java

brianb replied:
So, the only place suffix is used is in StringQueue and it is checked for null, so we do no string concatenation with null.  I am comfortable using empty string instead.

xiaoerge replied:
Cool,  lets go with a default empty string.

brianb replied:
done

xiaoerge commented on src/main/java/com/xiaoerge/x12/message/MessageFormatException.java:
```diff
@@ -0,0 +1,5 @@
+package com.xiaoerge.x12.message;
+
+public class MessageFormatException extends Exception {
```
If removing the parametrized constructor for MessageFormat.java,  remove this class as well.

xiaoerge commented:
Test the parametrized MessageFormat constructor and it should be all good.

brianb commented:
So, you're saying keep the parameterized constructor, but add a unit test to the patch?

I have another patch in the queue for length validation when the repetition character is present but it depends on these changes.

xiaoerge commented:
Yes, I agree with your point of having that constructor.  We now need to unit test that case by constructing non default MessageFormat("ISA....").  Try more than one, get the edge cases too.

brianb commented:
Had a re-think on the parameterized constructor.  I think your initial instinct is correct.

I moved the ISA parsing logic to X12Parser.parseMessageFormat(), added setters to MessageFormat, removed the constructor, and the MessageFormatException class and added some unit testing in X12MessageBaseTest.

xiaoerge commented:
Thanks @brianb
